code/modules/multiz/mz_items.dm

```diff
@@ -0,0 +1,55 @@
+/obj/item/weapon/ladder_mobile
+	name = "mobile ladder"
+	desc = "A lightweight deployable ladder, which you can use to move up or down. Or alternatively, you can bash some faces in."
+	icon_state = "ladder01"
+	icon = 'icons/obj/structures.dmi'
+	throw_range = 3
+	force = 10
+
+/obj/item/weapon/ladder_mobile/proc/place_ladder(atom/A,mob/user)
+
+	if(istype(A, /turf/simulated/open))         //Place into open space
+		user.visible_message("<span class='warning'>[user] begins to lower the ladder into the hole.</span>")
+		var/turf/below_loc = GetBelow(A)
+		if (!below_loc || (istype(/turf/space,below_loc)))
+			user << "<span class='notice'>Why would you do that?! There is only infinite space there...</span>"
+			return
+		handle_action(A,user)
+		var/obj/structure/ladder/mobile/body/R = new(A)
+		var/obj/structure/ladder/mobile/base/D = new(A)
+		D.forceMove(below_loc)
+		R.target_down = D 
+		D.target_up = R
+
+		user.drop_item()
+		qdel(src)
+
+	if(istype(A, /turf/simulated/floor))        //Place onto Floor
+		user.visible_message("<span class='warning'>[user] begins deploying the ladder on the floor.</span>")
+		var/turf/upper_loc = GetAbove(A)
+		if (!upper_loc || !istype(upper_loc,/turf/simulated/open))
+			user << "<span class='notice'>There is something above. You can't deploy!</span>"
+			return
+		handle_action(A,user)
```

Comment:
This only stalls. You should make `handle_action` return 1 or 0 depending on failure, and have this code cancel if failure is recognized.

Otherwise, even if handle_action fails, this continues without a care in the world.